devolutions-agent/src/updater/mod.rs

```diff
@@ -240,10 +243,38 @@ async fn check_for_updates(product: Product, update_json: &UpdateJson) -> anyhow
             }
         }
         VersionSpecification::Specific(version) => {
-            if version != remote_version {
-                warn!(%product, %version, "Product uptate target version does not match available version on devolutions.net, skipping update");
+            if version == detected_version {
+                info!(%product, %detected_version, "Product is up to date, skipping update");
                 return Ok(None);
             }
+
+            // If the target version is not available on devolutions.net, try to guess the requested
+            // version MSI URL by modifying the detected version.
+            //
+            // TODO(@pacmancoder): This is a temporary workaround until we have improved productinfo
+            // database with multiple version information.
+            if version != remote_version {
+                let modified_url = try_modify_product_url_version(&product_info.url, remote_version, version)?;
+
+                // Quick check if the modified URL points to existing resource.
+                let response = reqwest::Client::builder().build()?.head(&modified_url).send().await?;
+                if !response.status().is_success() {
+                    warn!(
+                        %product,
+                        %version,
+                        %modified_url,
+                        "Modified product URL does not exist, skipping update"
+                    );
+                    return Ok(None);
+                }
```

Comment:
suggestion: Not necessarily a 404 status, so the wording should be changed.

```suggestion
                if let Err(error) = response.error_for_status() {
                    warn!(
                        %error,
                        %product,
                        %version,
                        %modified_url,
                        "Failed to access the product URL, skipping update"
                    );

                    return Ok(None);
                }
```